community/dbms/src/main/java/org/neo4j/commandline/dbms/StoreLockChecker.java

```diff
@@ -58,10 +59,24 @@ static Closeable check( Path databaseDirectory ) throws CannotWriteException
         {
             if ( Files.isWritable( lockFile ) )
             {
-                StoreLockChecker storeLocker = new StoreLockChecker( new DefaultFileSystemAbstraction(),
-                        databaseDirectory.toFile() );
-                storeLocker.checkLock();
-                return storeLocker;
+                StoreLockChecker storeLocker = new StoreLockChecker( new DefaultFileSystemAbstraction(), databaseDirectory.toFile() );
+                try
+                {
+                    storeLocker.checkLock();
+                    return storeLocker;
+                }
+                catch ( StoreLockException le )
+                {
+                    try
+                    {
+                        storeLocker.close();
```

Comment:
we even recently updated StoreLocker tests suite and they claim that it's not the case but if you can't take a look on them and find where it can be the case that would be great, otherwise those test claims we still have a lock